test/unit/test_util.c

```diff
@@ -126,10 +126,42 @@ bool test_initial_underscore(void) {
 	mu_end;
 }
 
+typedef struct {
+	const char *name;
+	R_REF_TYPE;
+} TypeTest;
+
+static TypeTest *r_type_test_new(const char *name) {
+	TypeTest *tt = R_NEW0 (TypeTest);
+	tt->name = name;
+	return tt;
+}
+
+static void r_type_test_free(TypeTest *tt) {
+	tt->name = "";
+	free (tt);
+}
+
+R_REF_FUNCTIONS(TypeTest, type_test, r_type_test_free);
+
+bool test_references(void) {
+	TypeTest *tt = r_type_test_new ("foo");
+	mu_assert_eq (tt->refcount, 0, "reference count issue");
+	r_type_test_ref (tt);
+	mu_assert_eq (tt->refcount, 1, "reference count issue");
+	r_type_test_unref (tt);
+	mu_assert_streq (tt->name, "foo", "typetest name should be foo");
+	mu_assert_eq (tt->refcount, 0, "reference count issue");
+	r_type_test_unref (tt); // tt becomes invalid
+	// cant test further because its freed
```

Comment:
You can test this if you remove the `free()` from `r_type_test_free()` and check for `tt->name`

Reply:
yeah that will be a bit fake but safe

Reply:
done